### PR TITLE
Add deal creation form

### PR DESCRIPTION
**Pull request by chiemerie-arum**

### CODE REVIEW

File: src/components/input/form-field.tsx [hunk 1]
```diff
@@ -9,11 +9,13 @@ export type FormFieldProps = {
 const FormField = (props: FormFieldProps) => {
```
Comment (narender56) [inline src/components/input/form-field.tsx:9]: Would be nice to have file name to be Components name `FormField.tsx`

File: src/pages/deals/deal-creation-form.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import Progress from '@/components/progress';
+
+const DealForm = () => {
+  const onSubmit = (values: any) => {
+    setTimeout(() => {
+      alert(JSON.stringify(values, null, 2));
+    }, 500);
+  };
+
+  const getDealForm = () => {
```
Comment (narender56) [inline src/pages/deals/deal-creation-form.tsx:15]: move these functions outside of component

Comment (chiemerie-arum) [reply src/pages/deals/deal-creation-form.tsx:15]: I thought we agreed to keep everything about a component inside the component

Comment (chiemerie-arum) [reply src/pages/deals/deal-creation-form.tsx:15]: So we don't have situations like the Login page having a different component outside it

Comment (narender56) [reply src/pages/deals/deal-creation-form.tsx:15]: Keep it in the same file and move it outside the component

Comment (chiemerie-arum) [reply src/pages/deals/deal-creation-form.tsx]: Done

**Pull request merged**